Agents/FenlandTrajectoryAgent/agent/datainstantiation/cleaning_tool.py

```diff
@@ -0,0 +1,46 @@
+import os
+import re
+import pandas as pd
+import logging
+
+logger = logging.getLogger(__name__)
+
+def clean_gps_data(file_path, output_dir=None):
+    """
+    Reads a CSV file, extracts numeric values (removing any units), and returns the cleaned DataFrame.
+    If output_dir is provided, saves the cleaned DataFrame as a temporary file.
+    """
+    logger.info("Starting cleaning of GPS data from file: %s", file_path)
+    try:
+        df = pd.read_csv(file_path)
+        logger.info("CSV file loaded successfully with shape: %s", df.shape)
+    except Exception as e:
+        logger.error("Error loading CSV file %s: %s", file_path, e, exc_info=True)
+        raise e
+
+    def extract_numeric(value):
+        if isinstance(value, str):
+            matches = re.findall(r"[-+]?(?:\d*\.*\d+)", value)
```

Comment:
This looks over-zealous to me. Isn't this e.g. going to destroy any number in scientific notation?